### PR TITLE
Update to openapi spec 1.10

### PR DESCRIPTION
**Pull request by bdemers**
- Add IT to validate updating a users password with `strict` set
- Add setGroups(String... groupIds) to UserBuilder (simplifies testing)
- Update policies created in ITs to scope to users of a group (so tests can run in parallel)
- abstract randomGroup and random*Policy IT logic to ITSupport class

### CODE REVIEW

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/PasswordPoliciesIT.groovy [hunk 1]
```diff
@@ -16,19 +16,29 @@
...
 
-class PasswordPoliciesIT implements CrudTestSupport {
+class PasswordPoliciesIT extends ITSupport implements CrudTestSupport {
 
     @Override
     def create(Client client) {
+        Group group = randomGroup()
+
         return client.createPolicy(client.instantiate(PasswordPolicy)
+            .setConditions(client.instantiate(PasswordPolicyConditions)
```
Comment (laura-rodriguez) [inline integration-tests/src/test/groovy/com/okta/sdk/tests/it/PasswordPoliciesIT.groovy]: Is the indentation correct? It looks weird in this view.

Comment (bdemers) [reply integration-tests/src/test/groovy/com/okta/sdk/tests/it/PasswordPoliciesIT.groovy]: yup, it's pretty common when chaining methods together (i'd still love to figure out how to make this less verbose, but that is a different topic 😄)

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/ITSupport.groovy [hunk 2]
```diff
@@ -77,4 +86,49 @@ abstract class ITSupport implements ClientProvider {
 
         return user
     }
+
+    Group randomGroup(String name = "group-${UUID.randomUUID().toString()}") {
```
Comment (laura-rodriguez) [inline integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/ITSupport.groovy]: I think we talked about this before, but don't you have to include an SDK prefix in your random names? Something like "java-sdk"?

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/UsersIT.groovy [hunk 3]
```diff
@@ -250,7 +259,67 @@ class UsersIT extends ITSupport implements CrudTestSupport {
...
+                .buildAndCreate(client)
+        registerForCleanup(user)
+        validateUser(user, firstName, lastName, email)
+
+        // 2. Change the user's password
+        def request = client.instantiate(ChangePasswordRequest)
+            .setOldPassword(client.instantiate(PasswordCredential).setValue('Abcd1234'.toCharArray()))
+            .setNewPassword(client.instantiate(PasswordCredential).setValue('1234Abcd'.toCharArray()))
+
+        //
```
Comment (laura-rodriguez) [inline integration-tests/src/test/groovy/com/okta/sdk/tests/it/UsersIT.groovy]: Missing a comment here?

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/UsersIT.groovy [hunk 4]
```diff
@@ -250,7 +259,67 @@ class UsersIT extends ITSupport implements CrudTestSupport {
         // 2. Change the user's password
         UserCredentials credentials = user.changePassword(client.instantiate(ChangePasswordRequest)
             .setOldPassword(client.instantiate(PasswordCredential).setValue('Abcd1234'.toCharArray()))
-            .setNewPassword(client.instantiate(PasswordCredential).setValue('1234Abcd'.toCharArray())))
+            .setNewPassword(client.instantiate(PasswordCredential).setValue('1234Abcd'.toCharArray())), true)
+        assertThat credentials.getProvider().getType(), equalTo(AuthenticationProviderType.OKTA)
+
+        // 3. make the test recording happy, and call a get on the user
+        // TODO: fix har file
```
Comment (laura-rodriguez) [inline integration-tests/src/test/groovy/com/okta/sdk/tests/it/UsersIT.groovy]: Is this TODO still valid?

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/ITSupport.groovy (hunk 2 above)
Comment (bdemers) [reply integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/ITSupport.groovy]: +1, i'll try to add this, it doesn't work for all fields, as some of them have a shorter name description, but it's easy to try!

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/UsersIT.groovy (hunk 4 above)
Comment (bdemers) [reply integration-tests/src/test/groovy/com/okta/sdk/tests/it/UsersIT.groovy]: yes, but... i'll start another thread about killing this, i haven't been using those har files.

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/ITSupport.groovy (hunk 2 above)
Comment (bdemers) [reply integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/ITSupport.groovy]: groups worked, policy's were limited to 50 chars, but i did add `-java-` in the name

**Pull request merged**